### PR TITLE
update cell_census package to use SOMA API 0.5.0a3

### LINKED ISSUE #62: cell_census deps for NumPy & Numba are wrong
The code requires `numpy >= 1.21`, but the project.toml fails to specify the minimum required version.  

The code also needs a recent numba, which requires numpy < 1.24

So, deps should be:
numba >= 0.55  # which supports numpy 1.21
numpy >= 1.21 <= 1.23.5

### PR DESCRIPTION
**Pull request by bkmartinjr**
Update `cell_census` package and demo notebooks to use the latest SOMA API, `tiledbsoma==0.5.0a3`, including:
* fluent read iterators
* experiment query

In addition, 
* dependencies for `numba` and `numpy` were updated to match actual requirements.  Fixes #62
* small change to readme
* re-ran all notebooks

### CODE REVIEW

**Review by pablo-gar — APPROVED** [review]
I didn't see anything wrong and I tested it a bit locally, everything seems to be working.

File: api/python/cell_census/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # CELLxGENE Cell Census
 
-The `cell_census` package provides an API to facilitate use of the CZI Science Cell Census.
+The `cell_census` package provides an API to facilitate use of the CZI Science Cell Census. The Cell Census is a versioned ocntainer for the single-cell data hosted at [CELLxGENE Discover](https://cellxgene.cziscience.com/).
```
Comment (atolopko-czi) [inline api/python/cell_census/README.md:3]: typo

File: api/python/cell_census/src/cell_census/presence_matrix.py [hunk 2]
```diff
@@ -38,17 +37,4 @@ def get_presence_matrix(
...
-
-    arrow_sparse_tensors = [t for t in presence.read_sparse_tensor((slice(None),))]
-    flat_arrays = [t.to_numpy() for t in arrow_sparse_tensors]
-    data = np.concatenate(tuple(t[0] for t in flat_arrays))
-    coords = np.concatenate(tuple(t[1] for t in flat_arrays))
-    presence_matrix = sparse.coo_matrix(
-        (data.flatten(), (coords.T[0].flatten(), coords.T[1].flatten())), shape=presence.shape
-    ).tocsr()
-    return presence_matrix
+    return presence.read((slice(None),)).csrs().concat().to_scipy()
```
Comment (atolopko-czi) [inline api/python/cell_census/src/cell_census/presence_matrix.py:40]: 👏

File: api/python/notebooks/analysis_demo/comp_bio_census_info.ipynb [hunk 3]
```diff
@@ -1310,11 +1310,11 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.10.6"
+   "version": "3.9.16"
```
Comment (atolopko-czi) [inline api/python/notebooks/analysis_demo/comp_bio_census_info.ipynb:1313]: nit: should we stick a specified python version across all notebooks?

**Review by atolopko-czi — COMMENTED** [review]
Reviewed post-merge, LGTM.

File: api/python/cell_census/README.md (hunk 1 above)
Comment (atolopko-czi) [reply api/python/cell_census/README.md:3]: fixed on main

File: api/python/notebooks/analysis_demo/comp_bio_census_info.ipynb (hunk 3 above)
Comment (bkmartinjr) [reply api/python/notebooks/analysis_demo/comp_bio_census_info.ipynb:1313]: Can you elaborate the benefit of doing so?  It seems like we would have to manually enforce this convention (i.e., all of us just remember to use a given version)?

Comment (atolopko-czi) [reply api/python/notebooks/analysis_demo/comp_bio_census_info.ipynb:1313]: Probably not an issue if all notebooks are compliant with the lowest supported Cell Census Python version.

Comment (bkmartinjr) [reply api/python/notebooks/analysis_demo/comp_bio_census_info.ipynb:1313]: There are some protections in place, in `formatting.yml`, which lints against Py 3.7.  This isn't a 100% guarantee, but it helps. If you see more we can do, LMK and I'll get it into an issue

**Pull request merged**